### ISSUE TITLE
New child menu entries are not added automatically

### ISSUE THREAD

**Issue by ThiefMaster** [indico/indico#2049]
The logic in `menu_entries_for_event` doesn't take the case into account where the top-level menu entry already exists but a child entry is missing.

This is not a big issue for now, but it could be come an issue when we add a new child item for an existing menu entry in a future version.

**Comment by ThiefMaster** [comment]
```python
entry_names = {entry.name for entry in entries}
new_entry_names = signal_entries.viewkeys() - entry_names
```

Since `entries` only contains top-level entries, this currently contains more entries than necessary (all child elements even if they exist). Since the condition to add a new menu item also checks `data.parent is None` this doesn't cause problems but it's not very pretty.

**Comment by ThiefMaster** [comment]
I fixed this last year in d26b8a775
